avm/res/devops-infrastructure/managed-devops-pools/tests/e2e/waf-aligned/main.tests.bicep

```diff
@@ -0,0 +1,113 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-di.mdp-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'mdpwaf'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ========= //
+// Variables //
+// ========= //
+var azureDevOpsOrganizationName = 'contoso'
+var azureDevOpsProjectName = 'ContosoProject'
+
+// ============ //
+// Dependencies //
+// ============ //
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    devCenterName: 'dep-${namePrefix}-dc-${serviceShort}'
+    devCenterProjectName: 'dep-${namePrefix}-dcp-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+  }
+}
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      agentProfile: {
+        kind: 'Stateless'
+        resourcePredictions: {
+          timeZone: 'Central Europe Standard Time'
+          daysData: [
+            {
+              '09:00:00': 1
+              '17:00:00': 0
+            }
+            {}
+            {}
+            {}
+            {}
+            {}
+            {}
```

Comment:
Intended?

Reply:
Yes, this is intended for now. Each `{}` represents a day of the week which can hold two dictionary values 

```bicep
{
   startTime: <standby agents value>
   endTime: <standby agents value>
}
```

I yet have to think about how to make this easy for the user to fill in....

Reply:
I guess then we should at least add some comments :D 

That being said, does this also mean you have to set the value, or will it auto-scale if it's not set?

Reply:
Hey @AlexanderSehr,

For this iteration a description or metadata example will be the way to go.

I have experimented with a solution that accepts straight-forward user input, but cannot get it to work since I have to merge it with the property `agentProfile` at some point. I can't figure out **how** to do it at the moment.

```bicep
param parDay dayType = {
    Monday: {
      endAgents: 5
      endTime: '17:00'
      startAgents: 5
      startTime: '09:00'
    }
    Tuesday: {
      endAgents: 5
      endTime: '17:00'
      startAgents: 5
      startTime: '09:00'
    }
    Friday: {
      endAgents: 15
      endTime: '23:00'
      startAgents: 15
      startTime: '09:00'
    }
  }
  
var formattedDaysData = map(
  ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
  day =>
   contains(parDay, day) ? {
      '${parDay[day].startTime}': parDay[day].startAgents
      '${parDay[day].endTime}': parDay[day].endAgents
    } : {}
)
```

formattedDaysData outputs:
```bicep
[
  {
    '09:00': 5
    '17:00': 5
  }
  {
    '09:00': 5
    '17:00': 5
  }
  {}
  {}
  {
    '09:00': 15
    '23:00': 15
  }
  {}
  {}
]
```

This output is the required format, but somehow I have to merge it with `agentProfile`. If you have a clever solution I would be very interested in that 😄

Reply:
Happy to brainstorm some time. Just recently had the pleasure of messing with Bicep functions again.

That being said, I think you may have understood. I didn't mean to say that you should implement a type (though that would super helpful indeed), but rather that I'd suggest to add some comments into the test file so that somebody else who's looking at it also doesn't think these are leftover items, but has the epiphany that each item represents a day 😉
```bicep
[
  {} // I'm Monday
  {} // She's lying
  {} // ...
]
``` 
would do the trick 😄 Sure, it won't be in the readme which is unfortunate, but it's a start.

Reply:
For now I have added metadata for the object property + comments ;-)